ruby/ext/google/protobuf_c/extconf.rb

```diff
@@ -2,7 +2,9 @@
 
 require 'mkmf'
 
-$CFLAGS += " -O3 -std=c99 -Wno-unused-function -DNDEBUG "
+$CFLAGS += " -O3 -std=c99 -Wno-unused-function " +
+           "-Wno-declaration-after-statement -Wno-unused-variable " +
```

Comment:
Why did you have to disable all these warnings? The latest upb should be extremely warning-clean. Were you getting warnings?